### PR TITLE
PLANNER-2187 Use java 11 minimum for OptaPlanner 8

### PR DESCRIPTION
**Pull request by rsynek**
JIRA:
https://issues.redhat.com/browse/PLANNER-2187

Relates to:
https://github.com/kiegroup/optaplanner/pull/950
https://github.com/kiegroup/optaplanner-quickstarts/pull/6

### CODE REVIEW

File: download/upgradeRecipe/upgradeRecipe8.adoc [hunk 1]
```diff
@@ -17,25 +17,25 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
...
+For example on Fedora and RHEL:
++
+[source, bash]
+----
+sudo dnf install java-11-openjdk-devel
+----
+
+* On Windows and Mac, https://adoptopenjdk.net[download OpenJDK from AdoptOpenJDK].
+
+We currently intend to support a minimal version of Java 11 throughout the entire 8.x series.
```
Comment (triceo) [inline download/upgradeRecipe/upgradeRecipe8.adoc:38]: I'd remove this line. In general, let's not make promises years into an uncertain future.
Java LTS are not forever, anymore.

File: download/upgradeRecipe/upgradeRecipe8.adoc [hunk 2]
```diff
@@ -17,25 +17,25 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
...
+
+* On linux, get OpenJDK from your linux software repository.
+For example on Fedora and RHEL:
++
+[source, bash]
+----
+sudo dnf install java-11-openjdk-devel
+----
+
+* On Windows and Mac, https://adoptopenjdk.net[download OpenJDK from AdoptOpenJDK].
```
Comment (yurloc) [inline download/upgradeRecipe/upgradeRecipe8.adoc:36]: The most recent version of Apple's operating system for Mac computers is called [macOS](https://en.wikipedia.org/wiki/MacOS).

File: download/upgradeRecipe/upgradeRecipe8.adoc (hunk 1 above)
Comment (ge0ffrey) [reply download/upgradeRecipe/upgradeRecipe8.adoc:38]: We wrote a similar line in the 7.x upgrade recipe and it never was a problem.
The "currently intend" part is to make it clear it's not a solid promise.
Either way, I strongly believe we should not change the target version of the class files we produce in a minor or hotfix version. Which is basically what this is about.

I'd keep this sentence in, but I don't feel strongly about it. Radovan is the tie breaker I guess.

File: download/upgradeRecipe/upgradeRecipe8.adoc (hunk 2 above)
Comment (ge0ffrey) [reply download/upgradeRecipe/upgradeRecipe8.adoc:36]: Good point, we have a few other cases of this mistake that we should fix. Look for "Mac\W" regex in find in path.
@rsynek Do you mind taking that along when fixing this one or creating a separate jira for the backlog for it?

**Review by ge0ffrey — APPROVED** [review]

File: download/upgradeRecipe/upgradeRecipe8.adoc (hunk 1 above)
Comment (rsynek) [reply download/upgradeRecipe/upgradeRecipe8.adoc:38]: Neither do I understand the "We currently intend" as a promise.

File: download/upgradeRecipe/upgradeRecipe8.adoc (hunk 2 above)
Comment (rsynek) [reply download/upgradeRecipe/upgradeRecipe8.adoc]: Fix by an additional commit.

**Pull request merged**